tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {"errors": ["Active produce job not finished"]}
+            return
+
+        # update topic config
+        topics_cfg_keys = ["topic_prefix", "topic_count"]
+        forbidden = ["msg_rate", "msg_total"]
+        error_msgs = validate_keys(req.media, topics_cfg_keys, forbidden)
+        if len(error_msgs) > 0:
+            msg = f"Incoming request invalid: {', '.join(error_msgs)}\n" \
+                f"Valid keys are: {', '.join(topics_cfg_keys)}"
+            self.wlogger.error(msg)
+            resp.status = falcon.HTTP_400
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {"errors": msg}
+        else:
+            # Update topic configs
+            if 'topic_prefix' in req.media:
+                self.topics_config.topic_prefix = req.media['topic_prefix']
+            if 'topic_count' in req.media:
+                self.topics_config.topic_count = req.media['topic_count']
+            # start consumers
+            self.init_consumers()
+            self.consume(wait=False)
+
+            resp.status = falcon.HTTP_200
+            resp.content_type = falcon.MEDIA_TEXT
+            resp.text = "OK"
+
+    def terminate(self):
+        self.wlogger.debug("Terminating web consume class")
+        for t in self.topics.values():
+            t.terminate = True
+        return
+
+
+def start_webserver():
+    def terminate_handler(signum, frame):
+        """SIGTERM handler
+
+        Args:
+            signum (_type_): _description_
+            frame (_type_): _description_
+        """
+        signame = signal.Signals(signum).name
+        logger.info(f"{signame} ({signum}) received, terminating threads")
+        producer.terminate()
+        consumer.terminate()
+
+    def add_route(route: str, handler: AppCfg | StreamVerifier):
+        logger.debug(f"Registering handler for '/' as {type(handler)}")
+        app.add_route(route, handler)
+
+    global app_config
+    app = falcon.App()
+    logger = setup_logger(LOGGER_STARTUP)
+
+    # Add subpages
+    logger.debug("Initializing producer class")
+    producer = StreamVerifierProduce(app_config)
+    logger.debug("Initializing consumer class")
+    consumer = StreamVerifierConsume(app_config)
+    add_route("/", app_config)
+    add_route("/produce", producer)
+    add_route("/consume", consumer)
+
+    # Create and run service
+    with make_server('', app_config.web_port, app) as httpd:
+        # Serve until process is killed, SGTERM received or Keyboard interrupt
+        try:
+            logger.debug("Registering SIGTERM")
+            signal.signal(signal.SIGTERM, terminate_handler)
+            logger.info(f'Serving on port {app_config.web_port}...')
+            httpd.serve_forever()
+        except KeyboardInterrupt:
+            logger.info("Got keyboard interupt, exiting")
+            producer.terminate()
+            consumer.terminate()
```

Comment:
can we consolidate all cleanup into single shutdown hook? (regardless of exception or sigterm etc).

Reply:
Was thinking the same thing. Will update while working on atomic thing